src/app/pages/credentials/certificates-dash/certificates-dash.component.ts

```diff
@@ -249,52 +279,80 @@ export class CertificatesDashComponent implements OnInit {
   }
 
   certificateActions(): AppTableAction[] {
-    this.downloadActions = [{
-      icon: 'save_alt',
-      name: 'download',
-
-      onClick: (rowinner: Certificate) => {
-        const path = rowinner.CSR ? rowinner.csr_path : rowinner.certificate_path;
-        const fileName = rowinner.name + '.crt'; // what about for a csr?
-        this.ws.call('core.download', ['filesystem.get', [path], fileName]).pipe(untilDestroyed(this)).subscribe(
-          (res) => {
-            const url = res[1];
-            const mimetype = 'application/x-x509-user-cert';
-            this.storage.streamDownloadFile(this.http, url, fileName, mimetype)
-              .pipe(untilDestroyed(this))
-              .subscribe((file) => {
-                this.storage.downloadBlob(file, fileName);
-              }, (err) => {
-                this.dialogService.errorReport(helptext_system_certificates.list.download_error_dialog.title,
-                  helptext_system_certificates.list.download_error_dialog.cert_message, `${err.status} - ${err.statusText}`);
-              });
-          },
-          (err) => {
-            new EntityUtils().handleWSError(this, err, this.dialogService);
-          },
-        );
-        const keyName = rowinner.name + '.key';
-        this.ws.call('core.download', ['filesystem.get', [rowinner.privatekey_path], keyName]).pipe(untilDestroyed(this)).subscribe(
-          (res) => {
-            const url = res[1];
-            const mimetype = 'text/plain';
-            this.storage.streamDownloadFile(this.http, url, keyName, mimetype)
-              .pipe(untilDestroyed(this))
-              .subscribe((file) => {
-                this.storage.downloadBlob(file, keyName);
-              }, (err) => {
-                this.dialogService.errorReport(helptext_system_certificates.list.download_error_dialog.title,
-                  helptext_system_certificates.list.download_error_dialog.key_message, `${err.status} - ${err.statusText}`);
-              });
-          },
-          (err) => {
-            new EntityUtils().handleWSError(this, err, this.dialogService);
-          },
-        );
-        event.stopPropagation();
+    this.downloadActions = [
+      {
+        icon: 'save_alt',
+        name: 'download',
```

Comment:
This is probably out of scope. Would you add matTooltip for the download button, please?

https://user-images.githubusercontent.com/351613/127840957-730665fa-c468-4f4a-8f09-ffb0dfc5984f.mp4

Reply:
Sure :)